pywbem_mock/_baseprovider.py

```diff
@@ -493,3 +493,42 @@ def _get_bare_instance(instance_name, instance_store, copy=None):
             return instance_store.get(instance_name, copy=copy)
 
         return None
+
+    def is_subclass(self, subclass, superclass, class_store):
+        """
+        Return boolean indicating whether subclass is a (direct or indirect)
+        subclass of superclass.
+
+        This test is done by starting at the subclass and walking its
+        superclasses up to the root, so it is not as expensive as determining
+        all subclasses of a given class.
+
+        Parameters:
+
+          subclass (:term:`string`): Class name of subclass.
+
+          superclass (:term:`string`): Class name of superclass.
+
+          class_store (:class:`~pywbem_mock.BaseObjectStore`):
+            Class store of the CIM repository to search for the classes.
+
+        Returns:
+
+            bool: Boolean indicating whether subclass is a (direct or indirect)
+            subclass of superclass.
+
+        Raises:
+            KeyError: Subclass or superclass does not exist in the class store.
+        """
+        if subclass.lower() == superclass.lower():
+            # Found it
+            return True
+        cls = class_store.get(subclass)
```

Comment:
This check should be moved to the begin, in order to also fulfil the promise of raising KeyError for non-existing classes if both classes are invalid but equal.

Reply:
DONE